wit/types.wit

```diff
@@ -0,0 +1,23 @@
+interface types {
+    // milliseconds is the u32 number of milliseconds
+    type milliseconds = u32;
+
+    /// Errors that can be returned from the locker
+    variant error {
+        /// not-acquired error indicates the lock was not acquired because it has already been 
+        /// acquired by another entity.
+        not-acquired(string),
+        /// already-released error indicates the lease for the lock has already expired.
+        lease-expired(string),
+        /// already-released error indicates the lease for the loack has already been released.
+        already-released(string),
+        /// unknown indicates an unexpected error occurred when acquiring or releasing a lock
+        unknown(string),
+    };
```

Comment:
Rather than `variant`s for errors, consider using `wasi-cloud error`. Like, here:
https://github.com/WebAssembly/wasi-keyvalue/blob/0bce169f659170379aa4aeceef0e83711671acb9/wit/types.wit#L30

`wasi-cloud-error` looks like so:
https://github.com/WebAssembly/wasi-keyvalue/blob/0bce169f659170379aa4aeceef0e83711671acb9/wit/deps/io/error.wit#L1-L34

Reply:
I actually think in this case a variant is much more descriptive and useful for a consumer. I don't mind if the inner error is a `wasi-cloud-error` resource instead of a `string`, but I do like keeping variants around